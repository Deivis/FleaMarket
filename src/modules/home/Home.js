import React from 'react';

import InConstruction from '../../components/InConstructionWarning';

import './Home.scss';

const Home = () => (
  <div className="home">
    <header className="home__header">
      <span className="home__title">Flea market</span>
    </header>
    <div className="home__content">
      <InConstruction />
    </div>
  </div>
);

export default Home;
