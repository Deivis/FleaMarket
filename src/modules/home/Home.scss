@import '../../styles/_variables';
.home {
  @include size(100%);

  display: flex;
  flex-direction: column;
  height: 100vh;
}

.home__header {
  display: flex;
  justify-content: center;
  align-items: center;
  background: $medium-grey;
  flex: 0 0 auto;
  height: 70px;
}

.home__title {
  color: white;
  padding: 1em;
  font-size: 2.5em;
  font-weight: 100;
}

.home__content {
  display: flex;
  flex-flow: row wrap;
  flex-grow: 1;
  max-height: calc(100% - 70px);
  justify-content: center;
}

@media screen and (max-width: 1100px) {
  .content__aside {
    flex-basis: 100%;
  }
}
