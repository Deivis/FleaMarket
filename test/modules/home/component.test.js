import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Home from '../../../src/modules/home';
import configureStore from '../../../src/utils/configStore';

const store = configureStore();

describe('Home Tests', () => {
  test('Home Render', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Home />
          </Router>
        </Provider>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
