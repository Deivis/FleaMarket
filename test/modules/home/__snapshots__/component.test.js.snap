// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Tests Home Render 1`] = `
<div
  className="home"
>
  <header
    className="home__header"
  >
    <span
      className="home__title"
    >
      Flea market
    </span>
  </header>
  <div
    className="home__content"
  >
    <div
      className="container__in-constuction"
    >
      <span
        className="warning__in-constuction"
      >
         In construction 
      </span>
    </div>
  </div>
</div>
`;
